group/assignment6/move_cars.py

```diff
@@ -0,0 +1,38 @@
+def car_positions(state):
+    """ given state returns array where ith element is the position of ith car """
+    positions = [0] * len(start)
+    for place, car in enumerate(start):
+        positions[car] = place
+    return positions
+
+
+def generate_moves(start, final):
+    """ ith element of start state is the car currently in the ith space
+    ith element of final state is the car that should be in ith space
+    empty space is 0
+    yields optimal sequence of moves in format (car number, old space, new space) """
+
+    pos = car_positions(start)
+    for car, place in enumerate(pos):
+        if final[place] == car:
+            continue
+
+        if car:  # this is a cycle fix
```

Comment:
I can't see, how car variable can become "None" or "False". So, for now, I don't know how this "if" statement can be not satisfied. Please, clarify, what are you planning to do to make this statement useful?

Reply:
Possibly this should be "if car == 0" to make it more readable. However since 0 is an empty space, or basically a nonexistent car, I prefer this way. I.e. it can be read as "does this car actually exist"